### ISSUE TITLE
Caching invalidation

### ISSUE THREAD

**Issue by WatershedXiaolan** [WatershedXiaolan/Xiaolan-s-Gitbook#267]
TL:DR
- balance between write and read 

### How
Write-through cache:  
- data is written into the cache and the corresponding database simultaneously; 
- pro: complete data consistency between cache and disk 
- con: write operation can be slow; cache and flood 
- good for read-heavy applications 

Write-around cache
- data is written into disk directly, bypassing the cache 
- Pro: reduce the cache being flooded 
- Cons: cache miss for recent write data 
- good for write-heavy 

Write-back cache
- write to cache and write to disk at a later time 
- low latency and high throughput, but risk with data loss
